include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type
```

Comment:
It appears that `isnan` was [added to the std library for C++11][1], so this issue can only appear when using a std library implementation that is new enough to provide C++11 support, but old enough that the C++11 standard was not yet finalized.  I suppose it's possible that old versions of `stdlibc++` could suffer from this issue, but I don't know.

[1]: http://en.cppreference.com/w/cpp/numeric/math/isnan